### ISSUE TITLE
[BUG] npx git+ssh:// is not working

### ISSUE THREAD

**Issue by sagargurtu** [npm/cli#6212]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

Prerequisites: Clear npm cache first (see Steps To Reproduce below)

Running `npx -y git+ssh://git@github.com:prettier/prettier.git` throws without any error message.

However, `npm exec -y git+ssh://git@github.com:prettier/prettier.git` works successfully.

### Expected Behavior

Both of the above commands should produce the same output and not throw any errors.

### Steps To Reproduce

1. Run `rm -rf ~/.npm/_cacache`
2. Run `rm -rf ~/.npm/_npx`
3. Run `npx -y git+ssh://git@github.com:prettier/prettier.git` in any directory.
4. Process throws without displaying any error message.
5. Run `npm exec -y git+ssh://git@github.com:prettier/prettier.git`.
6. Command works successfully.
7. Subsequent reruns of `npx -y git+ssh://git@github.com:prettier/prettier.git` work now.

### Environment

- npm: 9.5.0
- Node.js: 18.14.2
- OS Name: macOS 13.2.1 
- System Model Name: Macbook Pro
- npm config:
```ini
; node bin location = /Users/bob/.nvm/versions/node/v18.14.2/bin/node
; node version = v18.14.2
; npm local prefix = /Volumes/projects
; npm version = 9.5.0
; cwd = /Volumes/projects
; HOME = /Users/bob
; Run `npm config ls -l` to show all defaults.
```

**Comment by fritzy** [comment]
duplicate of #5957
